azure-pipelines-templates/download-install-ncs-zephyr-toolchain.yml

```
@@ -0,0 +1,163 @@
+# Copyright (c) .NET Foundation and Contributors
+# See LICENSE file in the project root for full license information.
+
+parameters:
+  - name: ncsVersion
+    type: string
+    default: '2.0.0'
+  - name: zephyrSdkVersion
+    type: string
+    default: '0.15.1'
+
+steps:
+
+  # Use Python Version 3.10 and add it to path
+  - task: UsePythonVersion@0
+    displayName: Set Python to v3.10
+    inputs:
+      versionSpec: '3.10'
+      addToPath: true
+
+  - task: Cache@2
+    displayName: Cache latest NCS
+    inputs:
+      key: 'ncs-2_0_0 | ncsUpdateKey'
+      restoreKeys: ncsUpdateKey
+      path: $(Agent.TempDirectory)\ncs
+      cacheHitVar: NCS_CACHE_RESTORED
+
+  - task: PowerShell@2
+    condition: >-
+      and(
+        succeeded(),
+        ne(variables.NCS_CACHE_RESTORED, 'true')
+      )
+    displayName: Install pre-requisites for NCS
+    inputs:
+        targetType: 'inline'
+        script: |
+          choco feature enable -n allowGlobalConfirmation
+          choco install gperf dtc-msys2
+          pip3 install west
+
+  - task: PowerShell@2
+    condition: >-
+      and(
+        succeeded(),
+        ne(variables.NCS_CACHE_RESTORED, 'true')
+      )
+    displayName: Downloading NCS
+    inputs:
+        targetType: 'inline'
+        script: |
+          md $(Agent.TempDirectory)\ncs
+          cd $(Agent.TempDirectory)\ncs
+          west init -m https://github.com/nrfconnect/sdk-nrf --mr v${{ parameters.ncsVersion }}
+          west update
```

```suggestion
          west update --narrow -o=--depth=1
```

Will lead to shorter download times...
Source: https://raw.githubusercontent.com/coderbyheart/fw-nrfconnect-nrf-docker/saga/Dockerfile